code/dynamic_programming/src/Optimal_Binary_Search_Tree/Optimal_BST.cpp

```diff
@@ -0,0 +1,68 @@
+#include <vector>
+#include <iostream>
+
+using std::vector;
+using std::endl;
+using std::cin;
+using std::cout;
+/* 
+ * Given an array of keys and their respective frequencies, we want the cost of the Optimal Binary Search Tree
+ * The standard dynamic programming implementation will give us a solution with runtime complexity of O(n ^ 3)
+ */
+
+constexpr int infinity = 1e9;
+
+int min(int a, int b){ return ( a < b ? a : b); }
+
+int optimalSearchTree(const vector<int>& keys, const vector<int>& frequency, int totalNumberOfKeys) {
+    vector< vector<int> > cost( totalNumberOfKeys, vector<int>(totalNumberOfKeys, 0) ); 
+
+    vector<int> accumulatedFrequency(totalNumberOfKeys); // this vector will be used to store the prefix sum for the frequencies
+
+    accumulatedFrequency[0] = frequency[0];
+
+    for(int i = 1; i < totalNumberOfKeys; ++i) {
```

Comment:
Don't forget space after for, if ...etc. This is available for all the file. Avoid also putting braces when the loop or the condition have only one line.